exchange/utils.go

```diff
@@ -847,6 +859,49 @@ func mergeBidderRequests(allBidderRequests []BidderRequest, bidderNameToBidderRe
 	return allBidderRequests
 }
 
+func gdprFromGPP(r *openrtb2.BidRequest, gpp gpplib.GppContainer) {
+	if r.Regs != nil && r.Regs.GDPR == nil {
+		if len(r.Regs.GPPSID) > 0 {
+			// Set to 0 unless SID exists
+			r.Regs.GDPR = ptrutil.ToPtr[int8](0)
+			for _, id := range r.Regs.GPPSID {
+				if id == int8(gppConstants.SectionTCFEU2) {
+					r.Regs.GDPR = ptrutil.ToPtr[int8](1)
+				}
+			}
+		}
+	}
+
+	if r.User == nil || len(r.User.Consent) == 0 {
+		for i, id := range gpp.SectionTypes {
+			if id == gppConstants.SectionTCFEU2 {
+				if r.User == nil {
+					r.User = &openrtb2.User{}
+				}
+				r.User.Consent = gpp.Sections[i].GetValue()
+				return
+			}
+		}
+	}
+
+}
+func privacyFromGPP(r *openrtb2.BidRequest, gpp gpplib.GppContainer) {
+	if r.Regs != nil && len(r.Regs.USPrivacy) == 0 {
+		if len(r.Regs.GPPSID) > 0 {
```

Comment:
The length check is not necessary. The range iterator gracefully handles nil or empty arrays.